app/controllers/profile_controller.rb

```diff
@@ -0,0 +1,52 @@
+require_relative './application_controller'
+require_relative '../models/user'
+
+# Handles user profile.
+class ProfileController < ApplicationController
+  get '/' do
+    redirect '/' unless session[:email]
+
+    @css_link = 'style.css'
+    @user = User.find_by(email: session[:email])
+    session[:user]['image'] = @user.image
+    haml :profile
+  end
+
+  post '/update_user' do
+    if params[:user]['image']
+      file = params[:user]['image']
+      file_name = file[:filename]
+      temp_file = file[:tempfile]
+
+      File.open("./public/images/#{file_name}", 'wb') do |f|
+        f.write(temp_file.read)
+      end
+
+      params[:user]['image'] = file_name
+    end
+
+    User.where(email: session[:email]).update(params[:user])
+    session[:user] = params[:user]
+    session[:success] = 'Successfully Saved'
+
+    redirect '/profile'
+  end
+
+  post '/change_password' do
+    user = User.find_by(email: session[:email])
```

Comment:
prone to SQL injection

if someone enters `'; drop users;'` as an input, your application would be in a rather interesting state.

My advise: sanitize all inputs

Reply:
I have created a helper to escape_html